runtime/lua/vim/treesitter/languagetree.lua

```diff
@@ -263,7 +278,10 @@ end
 
 --- Returns whether this LanguageTree is valid, i.e., |LanguageTree:trees()| reflects the latest
 --- state of the source. If invalid, user should call |LanguageTree:parse()|.
----@param exclude_children boolean|nil whether to ignore the validity of children (default `false`)
+---@param exclude_children boolean|nil whether to ignore the validity of children (default `false`).
+--- `is_valid(false)` is sound but not complete, i.e.,
+--- * if it returns `true`, this LanguageTree is actually valid; but
+--- * even if this LanguageTree is actually valid, it may return `false`.
```

Comment:
This documenting bugs and defining it has a part of the API. 

`is_valid(false)` should always return true if the languagetree is valid.

Reply:
making it complete is possible but that would incur quite a bit of overhead (see note below), and i thought that kinda defeats the purpose of this function (quickly and soundly check necessity of `parse(true)`). i will implement it when i return

Reply:
fixed by running full injection in `is_valid(false)`

https://github.com/neovim/neovim/pull/26827/commits/3d1560bc8101a6e6cc4d641f7726bb1a4d1c951f#diff-70b0f3fa7d6baf9e605400a9f0e10431183df821ec4096e8097eaa5f4d5531baR310-R314